File: cow-chat-api/src/main/java/com/sparklecow/cowchat/message/MessageResponseDto.java
package com.sparklecow.cowchat.message;

import java.time.LocalDateTime;
import java.util.List;

public record MessageResponseDto(
        String id,
        String content,
        MessageType messageType,
        String filePath,
        String senderId,
        List<String> recipientIds,
        LocalDateTime timestamp,
        boolean delivered,
        boolean read
) {}

File: cow-chat-ui/src/app/models/message-response-dto.ts
import { MessageType } from "./message-request-dto";
export interface MessageResponseDto {
  id: string;
  content: string;
  messageType: MessageType;
  filePath?: string | null;
  chatId: string;
  senderId: string;
  recipientIds: string[];
  timestamp: string;
  delivered: boolean;
  read: boolean;
}
